bls/signature.go

```diff
@@ -0,0 +1,243 @@
+//go:build relic
+
+package bls
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	flowCrypto "github.com/onflow/flow-go/crypto"
+	"strings"
+)
+
+const SigningAlgorithm = flowCrypto.BLSBLS12381
+
+// TODO: what should the domainTag param be?
+var HashingAlgorithm = flowCrypto.NewExpandMsgXOFKMAC128("deso-protocol")
+
```

Comment:
Leave detailed comments for all of this stuff please.

Reply:
In your comment, explain how large the resulting aggregated signature ends up being. For example, if I have N signatures that each signed a distinct payload, is the resulting aggregated signature larger than if the signatures all signed the same payload?

Reply:
Hmm... reading the other code, it seems unclear as to whether the signatures all needed to sign the same payload or whether they all can sign distinct payloads. Comment would help here.

Reply:
Ok, I see now that this libary is designed to support only the verification of a single payload, which makes sense. Still let's leave a lot of comments here so it's all more clear.